File: Data Structures & Algorithms/k-closest-points-to-origin/submission-0.java
class Solution {

    private int distance(int[] point){

        return point[0]*point[0]+point[1]*point[1];
    }

    public int[][] kClosest(int[][] points, int k) {
        
        PriorityQueue<int[]> heap = new PriorityQueue<>((a,b)-> Integer.compare(distance(a), distance(b)));

        for (int[] point : points){
            heap.offer(point);
        }

        int[][] result = new int[k][2];

        for (int i=0; i<k;i++){
            result[i]= heap.poll();
        }
        return result;
    }
}

File: Data Structures & Algorithms/longest-common-prefix/submission-0.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        result = ""

        for i in range(len(strs[0])):
            char = strs[0][i]
            result+=char
            for string in strs:
                if len(string)>i and string[i]==char:
                    continue
                else:
                    result=result[:-1]
                    return result
                
        return result